src/MediaReplacer.php

```diff
@@ -0,0 +1,324 @@
+     * @param string $new_file_path The path to the new file.
+     */
+    private function replace_media_file(int $old_file_id, string $new_file_path): bool
+    {
+        try {
+            // Get the old file path
+            $old_file_path = get_attached_file($old_file_id);
+
+            // Check if the old file exists
+            if (file_exists($old_file_path)) {
+                unlink($old_file_path);
+            }
+
+            // Move the new file to the old file's location
+            $file_renamed = rename($new_file_path, $old_file_path);
+
+            // If the file was not renamed, abort
+            if (!$file_renamed) {
+                return false;
+            }
+
+            // Update the attachment metadata with new information
+            $filetype = wp_check_filetype($old_file_path);
+            $attachment_data = array(
+                'ID' => $old_file_id,
+                'post_mime_type' => $filetype['type'],
+                'post_title' => preg_replace('/\.[^.]+$/', '', basename($old_file_path)),
+                'post_content' => '',
+                'post_status' => 'inherit',
+            );
+
+            // Update the database record for the file
+            $post_updated = wp_update_post($attachment_data);
+
+            // If the post was not updated, abort
+            if (is_wp_error($post_updated) || $post_updated === 0) {
+                return false;
+            }
+
+            // Update file metadata
+            // By calling the "wp_update_attachment_metadata" function,
+            // the WP Stateless plugin syncs the file with Google Storage.
+            // https://github.com/udx/wp-stateless/blob/0871da645453240007178f4a5f243ceab6a188ea/lib/classes/class-bootstrap.php#L376
+            $attach_data = wp_generate_attachment_metadata($old_file_id, $old_file_path);
+            $post_meta_updated = wp_update_attachment_metadata($old_file_id, $attach_data);
+
+            // If the post meta was not updated, abort
+            return $post_meta_updated;
+        } catch (\Exception $e) {
+            set_transient('media_replacement_error', $e->getMessage(), 5);
+            return false;
+        }
+    }
+
+    /**
+     * Purge the Cloudflare cache for a specific URL.
+     *
+     * @param string $url The URL to be purged.
+     */
+    private function purge_cloudflare(string $url): void
+    {
+        try {
+            $cf = new CloudflarePurger();
+            $generator = $cf->purge([$url]);
+            $api_responses = [];
+
+            foreach ($generator as $purge_result) {
+                [$api_response] = $purge_result;
+                $api_responses[] = $api_response;
+            }
+
+            if ($api_responses[0]) {
+                $message = 'Cloudflare successfully purged URL: ' . $url;
+                set_transient('cloudflare_purge_message', $message, 5);
+            } else {
+                $error_message = 'Cloudflare could not purge URL: ' . $url;
```

Comment:
On a similar note: "There was an error purging the URL from cache"

I guess my point is that most editor don't necessarily know what Cloudflare is. But they do understand cache.

Reply:
Done!